### ISSUE TITLE
AudioKit.setOutputDevice not changing output

### ISSUE THREAD

**Issue by borg** [AudioKit/AudioKit#1546]
Changing output device on MacOS after AK started doesn't change eg. between Internal and External Soundcard (or Multiple outputs). Also doesn't throw error. Just zen-like silence. Is the code right?

```
    @objc func setOutput(_ v: Int) {
        do {
            try AudioKit.stop()
        } catch {
            AKLog("AudioKit did not start!")
        }
        do {
            print("setting output to \(AudioKit.outputDevices?[v])")
            try AudioKit.setOutputDevice((AudioKit.outputDevices?[v])!)
             do {
                try AudioKit.start()
            } catch {
                AKLog("AudioKit did not start!")
            }
        } catch{
            print("Failed to set output \(v)")
            for o in AudioKit.outputDevices! {
                print(o)
            }
        }
    }
```

Mojave, 
AK current git commit d3da07d24b43ce23fbb7a5bafbf43db4104e8211

**Comment by aure** [comment]
Related to https://github.com/AudioKit/AudioKit/issues/891 ?

**Comment by borg** [comment]
Not tested on iOS, but in addition, which may or may not be related, using Multi-output device on OSX gives error.

Multi-output Aggregate device 
```
Playground execution terminated: An error was thrown and was not caught:
Error Domain=com.apple.coreaudio.avfaudio Code=-10875 "(null)" UserInfo={false condition=IsFormatSampleRateAndChannelCountValid(outputHWFormat)}
```
Only true if default system output is not manually overridden before starting AK output.

**Comment by ryanfrancesconi** [comment]
I will look at this.

**Comment by ryanfrancesconi** [comment]
This works fine for me on macOS:

```
func setOutputDevice(name: String) {
    guard let outputs = AudioKit.outputDevices else { return }

    // look for the device
    guard let device = outputs.filter({ (theDevice) -> Bool in
        theDevice.name == name
    }).first else {
        AKLog("Unable to find", name, "in audio output devices.")
        return
    }

    do {
        try AudioKit.setOutputDevice(device)
        AKLog("Set output device to", name)

        if AudioKit.engine.isRunning {
            try AudioKit.stop()
            try AudioKit.start()
        }
    } catch {
        AKLog("Unable to set audio output to \(device.name)")
    }
}
```
![image](https://user-images.githubusercontent.com/2917795/50564836-5e3ef200-0cdd-11e9-913e-680f33b7d97e.png)

**Comment by aure** [comment]
Actually aggregate devices are not supported by AVAudioengine.  We've filed several radars with Apple only to have them closed as won't fix issues.

**Comment by aure** [comment]
Assuming this was the AVAudioEngine issue, I'm going to assume we can't really deal with it on our level and I'll close the issue unless I hear different. Thanks!

**Comment by YuantongL** [comment]
> Actually aggregate devices are not supported by AVAudioengine. We've filed several radars with Apple only to have them closed as won't fix issues.

@aure Is it still true that AVAudioEngine does not support aggregate device?

**Comment by aure** [comment]
@YuantongL I don't know.
